node/nodeBlocks.go

```diff
@@ -0,0 +1,58 @@
+package node
+
+import (
+	"encoding/hex"
+
+	apiBlock "github.com/ElrondNetwork/elrond-go/api/block"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/node/blockAPI"
+)
+
+// GetBlockByHash return the block for a given hash
+func (n *Node) GetBlockByHash(hash string, withTxs bool) (*apiBlock.APIBlock, error) {
+	decodedHash, err := hex.DecodeString(hash)
+	if err != nil {
+		return nil, err
+	}
+
+	apiBlockProcessor := n.createAPIBlockProcessor()
+	return apiBlockProcessor.GetBlockByHash(decodedHash, withTxs)
+}
+
+// GetBlockByNonce returns the block for a given nonce
+func (n *Node) GetBlockByNonce(nonce uint64, withTxs bool) (*apiBlock.APIBlock, error) {
+	apiBlockProcessor := n.createAPIBlockProcessor()
+
+	return apiBlockProcessor.GetBlockByNonce(nonce, withTxs)
+}
+
+func (n *Node) createAPIBlockProcessor() blockAPI.APIBlockHandler {
+	if n.shardCoordinator.SelfId() != core.MetachainShardId {
+		return blockAPI.NewShardApiBlockProcessor(
+			&blockAPI.APIBlockProcessorArg{
+				SelfShardID:              n.shardCoordinator.SelfId(),
+				Store:                    n.store,
+				Marshalizer:              n.internalMarshalizer,
+				Uint64ByteSliceConverter: n.uint64ByteSliceConverter,
+				HistoryProc:              n.historyProcessor,
+				UnmarshalTx: func(txBytes []byte, txType string) (*transaction.ApiTransactionResult, error) {
```

Comment:
Extract to function, then pass its pointer here?

Reply:
done.